### PR TITLE
#663 ensuring the inspect command uses the unit label if available

### PR DESCRIPTION
**Pull request by robons**
Satisfies issue #663

### CODE REVIEW

File: src/csvcubed/utils/csvdataset.py [hunk 1]
```diff
@@ -264,8 +367,9 @@ def transform_dataset_to_canonical_shape(
             unit_col_about_urls_value_urls,
             obs_val_col_titles_about_urls,
             col_names_col_titles,
+            data_cube_state,
         )
 
         canonical_shape_dataset = melted_df.drop("Observation Value", axis=1)
 
-    return (canonical_shape_dataset, measure_col, unit_col)
+    """
```
Comment (robons) [inline src/csvcubed/utils/csvdataset.py:375]: Can you remove this big comment now?

File: src/csvcubed/utils/csvdataset.py [hunk 2]
```diff
@@ -204,41 +296,52 @@ def transform_dataset_to_canonical_shape(
...
-                qube_components, csvw_metadata_json_path
-            )
-            canonical_shape_dataset[measure_col] = (
-                result.measure_label
-                if result.measure_label is not None
-                else result.measure_uri
-            )
-        else:
-            measure_col = measure_col_retrived
+        # unit_col_retrived = get_standard_shape_unit_col_name_from_dsd(qube_components)
```
Comment (robons) [inline src/csvcubed/utils/csvdataset.py:305]: Can you remove this comment now?

**Comment by robons** [comment]
@nimshi89 can you take another look at the code smells? It's picked up some relevant stuff.

File: src/csvcubed/utils/csvdataset.py [hunk 3]
```diff
@@ -181,6 +194,85 @@ def _melt_data_set(
     )
 
 
+def cube_in_standard_shape(
```
[the remark this answers is not in the document]
Comment (nimshi89) [reply src/csvcubed/utils/csvdataset.py:197]: done

File: src/csvcubed/utils/csvdataset.py [hunk 4]
```diff
@@ -181,6 +194,85 @@ def _melt_data_set(
...
+            if result.measure_label is not None
+            else result.measure_uri
+        )
+    else:
+        measure_col = measure_col_retrived
+
+    return (canonical_shape_dataset, measure_col, unit_col)
+
+
+def cube_in_pivoted_shape(
```
[the remark this answers is not in the document]
Comment (nimshi89) [reply src/csvcubed/utils/csvdataset.py:233]: done

File: src/csvcubed/utils/csvdataset.py (hunk 1 above)
Comment (nimshi89) [reply src/csvcubed/utils/csvdataset.py:375]: sorted

File: src/csvcubed/utils/csvdataset.py (hunk 2 above)
Comment (nimshi89) [reply src/csvcubed/utils/csvdataset.py:305]: sorted

**Review by nimshi89 — APPROVED** [review]
seems to be in order

**Pull request merged**